File: conversion_scripts/domain/races.py
"""Defines the domain of race definitions."""
from __future__ import annotations

from typing import Optional

from pydantic import Extra  # pylint: disable=unused-import
from pydantic import BaseModel, Field

from .books import Book


class Race(BaseModel, extra=Extra.forbid):
	"""Full definition of a race (with optional base race)."""

	name: str = Field(description="Name of the race.")
	source: Book = Field(description="Source book in which race is defined.")
	base_name: Optional[str] = Field(None, alias='baseName', description="Name of the base race.")
	base_source: Optional[Book] = Field(None, alias='baseSource', description="Source of the base race.")

File: src/data/races.js
/**
 * Data definition for races.
 */

/**
 * Full definition of a race (with optional base race).
 */
class Race {
    /**
     * @param {String} name_ Name of the race.
     * @param {Book} source Source book in which race is defined.
     * @param {String} base_name Name of the base race.
     * @param {Book} base_source Source of the base race.
     */
    constructor ({ name_, source, base_name = null, base_source = null }) {
        this.name_ = name_;
        this.source = source;
        this.base_name = base_name;
        this.base_source = base_source;
    }
};

export { Race };
